Owncloud iOs Client/Tabs/FileTab/Share/ShareMainViewController.m

```diff
@@ -332,237 +134,48 @@ - (void) setBarButtonStyle {
-    
-    DLog(@"SharesWith: %@", sharesWith);
-    
     [self.sharedUsersOrGroups removeAllObjects];
     [self.sharesOfFile removeAllObjects];
+    [self.sharedPublicLinks removeAllObjects];
     
-    for (OCSharedDto *shareWith in sharesWith) {
-        if (shareWith.shareType == shareTypeUser || shareWith.shareType == shareTypeGroup || shareWith.shareType == shareTypeRemote) {
+    self.sharesOfFile = [ManageSharesDB getSharesByUser:APP_DELEGATE.activeUser.idUser andPath:path];
+    
+    DLog(@"Number of Shares of file: %lu", (unsigned long)self.sharesOfFile.count);
+    
+    for (OCSharedDto *shareItem in self.sharesOfFile) {
+        
+        if (shareItem.shareType == shareTypeUser || shareItem.shareType == shareTypeGroup || shareItem.shareType == shareTypeRemote) {
+            
             
             OCShareUser *shareUser = [OCShareUser new];
-            shareUser.name = shareWith.shareWith;
-            shareUser.displayName = shareWith.shareWithDisplayName;
-            shareUser.sharedDto = shareWith;
-            shareUser.shareeType = shareWith.shareType;
+            shareUser.name = shareItem.shareWith;
+            shareUser.displayName = shareItem.shareWithDisplayName;
+            shareUser.sharedDto = shareItem;
+            shareUser.shareeType = shareItem.shareType;
             
             [self.sharedUsersOrGroups addObject:shareUser];
-            [self.sharesOfFile addObject:shareWith];
-        }
-    }
-    
-    self.sharedUsersOrGroups = [ShareUtils manageTheDuplicatedUsers:self.sharedUsersOrGroups]; 
- 
-}
-
-
-
-- (void) didSelectCloseView {
-    
-    [self dismissViewControllerAnimated:true completion:nil];
-}
-
-- (void) sharedLinkSwithValueChanged: (UISwitch*)sender {
-    
-    if (APP_DELEGATE.activeUser.hasCapabilitiesSupport == serverFunctionalitySupported && APP_DELEGATE.activeUser.capabilitiesDto) {
-        OCCapabilities *cap = APP_DELEGATE.activeUser.capabilitiesDto;
-        
-        if (!cap.isFilesSharingShareLinkEnabled) {
-            sender.on = false;
-            UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"not_share_link_enabled_capabilities", nil) message:@"" delegate:nil cancelButtonTitle:NSLocalizedString(@"ok", nil) otherButtonTitles:nil, nil];
-            [alertView show];
-            return;
-        }
-    }
-    
-    self.isShareLinkEnabled = sender.on;
-    
-    if (self.isShareLinkEnabled) {
-        [self getShareLinkView];
-    } else {
-        [self unShareByLink];
-    }
-}
-
-
-- (void) passwordProtectedSwithValueChanged:(UISwitch*) sender{
-    
-     if (self.isPasswordProtectEnabled){
-
-        if (APP_DELEGATE.activeUser.hasCapabilitiesSupport == serverFunctionalitySupported) {
-            OCCapabilities *cap = APP_DELEGATE.activeUser.capabilitiesDto;
             
-            if (cap.isFilesSharingPasswordEnforcedEnabled) {
```

Comment:
We should have both classes ```OCShareUser```  and ```OCShareLink```, or none of them (and just use ```OCSharedDto```) ; but not only ```OCShareUser```.

But, definitely, not for now.